### PR TITLE
Add contributing guidelines

### PR DESCRIPTION
**Pull request by demjened**
Add `CONTRIBUTING.md` with sections for notebooks and example apps, and reference it in `README.md`.

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+## Before you start
+
+Prior to opening a pull request, please:
+- Create an issue to [discuss the scope of your proposal](https://github.com/elastic/elasticsearch-labs/issues). We are happy to provide guidance to make for a pleasant contribution experience.
+- Sign the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/). We are not asking you to assign copyright to us, but to give us the right to distribute your code without restriction. We ask this of all contributors in order to assure our users of the origin and continuing existence of the code. You only need to sign the CLA once.
+
+## Contributing to Python notebooks 📒
+
+### Why
```
Comment (demjened) [inline docs/CONTRIBUTING.md:13]: I went with the [5W1H](https://www.wimi-teamwork.com/blog/the-5w1h-method-project-management-defined-and-applied) method to break down the list of bulletpoints into more concise chunks.

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+* The main goal of this repo is to help people learn about solving various problems with the Elastic Stack using step-by-step interactive guides and specific applications.
+* Remember your target audience: developers who want to try out some technology with Elastic. They may not be familiar with all the technologies.
+
+### Where
+
+* Select a folder under [notebooks](../notebooks/README.md) that matches the category of your notebook. If none of them match, create a new folder.
+* Create a folder under the category for your notebook.
```
Comment (leemthompo) [inline docs/CONTRIBUTING.md:21]: I wonder if creating a separate folder for every notebook isn't overkill? I'm leaning towards updating the README in the notebook category folder to summarize what each notebook does. WDYT?

**Review by leemthompo — APPROVED** [review]
This is great 🚀, I had one minor question

File: docs/CONTRIBUTING.md (hunk 2 above)
Comment (demjened) [reply docs/CONTRIBUTING.md]: Good point, I removed that bulletpoint and added one about updating the `README.md` file.

**Pull request merged**